### ISSUE TITLE
Tuple arg gets ignored when variable is assigned a value

### ISSUE THREAD

**Issue by michelbl** [microsoft/pyright#6208]
In the following code, the variable `test_tuple_any` loses its type hint `tuple[int]` when assigned a value of type `tuple[Any]`:

```python
from typing import Any

def returns_tuple_any() -> tuple[Any]:
    return (42,)


test_tuple_int: tuple[int] = returns_tuple_any()
# test_tuple_int now has the type tuple[Any]

print(test_tuple_int[0] + "a") # this passes pyright checks 😱
```

As a result, **pyright 1.1.332** does not report the type mismatch inside the `print` instruction.

However, there is no such behavior for lists, and the error is correctly reported:

```python
from typing import Any


def returns_list_any() -> list[Any]:
    return [42]


test_list_int: list[int] = returns_list_any()

print(test_list_int[0] + "a")
# error: Operator "+" not supported for types "int" and "Literal['a']" (reportGeneralTypeIssues) 😌 
```

**Comment by michelbl** [comment]
As a workaround, I currently `cast` my returned value to the expected type.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.333, which I just published. It will also be included in a future release of pylance.
